src/main/java/com/springbootassignment/cruddemo/service/CourseServiceImpl.java

```diff
@@ -0,0 +1,53 @@
+package com.springbootassignment.cruddemo.service;
+
+import com.springbootassignment.cruddemo.entity.Course;
+import com.springbootassignment.cruddemo.repository.CourseRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class CourseServiceImpl implements CourseService {
+
+    private CourseRepository courseRepository;
```

Comment:
Pls use @Autowired annotation instead of manually initializing the repo variable.
Pls update in other service and controllers too..